dev-tools/ls-apis/src/api_metadata.rs

```diff
@@ -186,6 +200,24 @@ impl TryFrom<RawApiMetadata> for AllApiMetadata {
     }
 }
 
+/// Describes how an API in the system manages drift between client and server
+#[derive(Default, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+#[serde(tag = "versioned_how", content = "versioned_how_reason")]
+pub enum VersionedHow {
+    /// We have not yet determined how this API will be versioned.
+    #[default]
+    Unknown,
+
+    /// This API will be versioned solely on the server.  (The update system
+    /// will ensure that servers are always updated before clients.)
+    Server,
+
+    /// This API will be versioned on the client.  (The update system cannot
+    /// ensure that servers are always updated before clients.)
+    Client(String),
```

Comment:
`Could this be `{ reason: String }`?

Reply:
I'm not sure how to do this while keeping the TOML format clean.  If you just change this like that, then you get:

```
$ cargo xtask ls-apis dag-check
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.51s
     Running `target/debug/xtask ls-apis dag-check`
   Compiling omicron-ls-apis v0.1.0 (/home/dap/omicron-merge/dev-tools/ls-apis)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 2.12s
     Running `target/debug/ls-apis dag-check`
Error: parse "/home/dap/omicron-merge/dev-tools/ls-apis/api-manifest.toml"

Caused by:
    TOML parse error at line 161, column 1
        |
    161 | [[apis]]
        | ^^^^^^^^
    invalid type: string "depends on itself (i.e., instances call each other)", expected struct variant VersionedHow::Client    
```

Right now, the tagging attribute here:

```rust
#[serde(tag = "versioned_how", content = "versioned_how_reason")]
```

means that in the file you put:

```
versioned_how = "client"
versioned_how_reason = "depends on itself (i.e., instances call each other)"
```

which I think is pretty nice?  I'm not sure how it would work with `{ reason: String }`.

Reply:
oh! yeah, that makes sense -- in that case just a comment would be ok. I was tripping up more on there not being a name for the field.

Reply:
I'll mention btw that serde(flatten) might become an issue in the future, due to how serde does internal buffering. For example, [serde-ignored](https://github.com/dtolnay/serde-ignored), which I think is often very valuable, doesn't work with flatten.

I've personally stopped using serde(flatten), and if it's never necessary for me I'll maintain two copies of the same type -- one for deserialization and one for internal use.

But it's not a blocker here, just something to keep in mind.